zk-sdk/src/encryption/elgamal.rs

```diff
@@ -414,6 +442,7 @@ impl ElGamalSecretKey {
     /// Deterministically derives an ElGamal secret key from a Solana signer and a public seed.
     ///
     /// See `ElGamalKeypair::new_from_signer` for more context on the key derivation.
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
nit: I can't comment further up, but similar to the last one, would it make more sense to have this whole impl block take the `#[cfg(not(target_arch = "wasm32"))]` attribute? And then you can put all of the functions that are ok for wasm in a separate impl block, which might look cleaner